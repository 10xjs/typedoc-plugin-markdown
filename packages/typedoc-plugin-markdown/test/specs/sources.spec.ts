import * as Handlebars from 'handlebars';

import { TestApp } from '../test-app';

describe(`Sources:`, () => {
  let testApp: TestApp;
  let partial: Handlebars.TemplateDelegate;
  beforeAll(() => {
    testApp = new TestApp(['sources.ts']);
    testApp.bootstrap();
    partial = TestApp.getPartial('member.sources');
  });

  test(`should display implementation of sources'`, () => {
    expect(
      TestApp.compileTemplate(
        partial,
        testApp.findReflection('SomeClass').findReflectionByName('prop'),
      ),
    ).toMatchSnapshot();
  });

  test(`should display inherited and overide sources'`, () => {
    expect(
      TestApp.compileTemplate(
        partial,
        testApp.findReflection('AnotherInterface').findReflectionByName('prop'),
      ),
    ).toMatchSnapshot();
  });
});
