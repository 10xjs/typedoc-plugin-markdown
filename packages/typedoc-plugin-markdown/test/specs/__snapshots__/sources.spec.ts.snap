// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sources: should display implementation of sources' 1`] = `
"*Implementation of [SomeInterface](_sources_.someinterface.md).[prop](_sources_.someinterface.md#prop)*

*Defined in sources.ts:6*
"
`;

exports[`Sources: should display inherited and overide sources' 1`] = `
"*Inherited from [SomeInterface](_sources_.someinterface.md).[prop](_sources_.someinterface.md#prop)*

*Overrides [SomeClass](../classes/_sources_.someclass.md).[prop](../classes/_sources_.someclass.md#prop)*

*Defined in sources.ts:2*
"
`;
